hyperspy/datasets/artificial_data.py

```diff
@@ -356,3 +363,231 @@ def get_atomic_resolution_tem_signal2d():
+        scale=0.54,
+        offset=222.495,
+        is_binned=False,
+        )
+
+    #Artificial luminescence peak
+    gaussian_peak = components1d.Gaussian(A=5000, centre=375, sigma=25)
+
+    #Creating data array, possibly with noise and baseline
+    data = gaussian_peak.function(nm_axis.axis)
+    if add_noise:
+        data += (random_state.uniform(size=len(nm_axis.axis)) - 0.5)*1.4
+    if add_baseline:
+        data += 350.
+
+    #Creating the signal with axis and data
+    sig = Signal1D(data,axes=[nm_axis])
+    sig.metadata.General.title = 'Artificial Luminescence Spectrum'
+
+    #if not uniform, transformation into non-linear axis
+    if not uniform:
+        hc = 1239.84198 #nm/eV
+
+        #eV axis creation. Note how the slice is inverted to have low energy first
+        evax = FunctionalDataAxis(expression="a/x",
+                                  x=sig.isig[::-1].axes_manager[0],
+                                  a=hc,
+                                  name='Energy',
+                                  units='eV',
+                                  navigate=False)
+
+        #Creating the signal, also with energy inverted
+        sig = Signal1D(data[::-1]*hc/evax.axis**2,axes=[evax])
+        sig.metadata.General.title = 'Artificial Luminescence Spectrum'
+
+    return sig
+
+get_luminescence_spectrum_nonuniform.__doc__ %= (ADD_BASELINE_DOCSTRING,
+                                                 ADD_NOISE_DOCSTRING)
+
+def get_luminescence_map_nonuniform(uniform=False, add_baseline=False, add_noise=True,random_state=None):
+    """Get an artificial luminescence 10-by-10 map in wavelength (nm, uniform)
+    or energy (eV, non-uniform) scale, simulating luminescence spectral maps
+    recorded with a diffracting spectrometer. Some random noise can also be
+    added to to simulate experimental noise.
+
+    Parameters
+    ----------
+    uniform: bool.
+        return uniform (wavelength) or non-uniform (energy) spectrum
+    %s
+    %s
+    random_state: None or int
+        initialise state of the random number generator
+
+    Example
+    -------
+    >>> import hyperspy.datasets.artifical_data as ad
+    >>> s = ad.get_luminescence_map_nonuniform()
+    >>> s.plot()
+
+    With constant baseline
+
+    >>> s = ad.get_luminescence_map_nonuniform(uniform=True, add_baseline=True)
+    >>> s.plot()
+
+    Make the noise the same for multiple spectra, which can
+    be useful for testing fitting routines
+
+    >>> s1 = ad.get_luminescence_map_nonuniform(random_state=10)
+    >>> s2 = ad.get_luminescence_map_nonuniform(random_state=10)
+    >>> (s1.data == s2.data).all()
+    True
+
+    See also
+    --------
+    get_low_loss_eels_signal,
+    get_core_loss_eels_signal,
+    get_low_loss_eels_line_scan_signal,
+    get_core_loss_eels_line_scan_signal,
+    get_core_loss_eels_model,
+    get_atomic_resolution_tem_signal2d,
+    get_luminescence_spectrum_nonuniform
+    """
+    from hyperspy.signals import Signal1D
```

Comment:
Unless strictly necessary, the import should be at the top of the file. Can you move them to the top?

Reply:
OK I did this by imitation of the other artificial datasets function. Will move it to the top.